integration/client_sdk/go/utils.go

```diff
@@ -97,3 +115,158 @@ func setupNetwork(channel string) (*gateway.Network, error) {
+		Label:   cc.Id,
+		Package: ccPkg,
+	}
+	resp, err := client.LifecycleInstallCC(installCCReq, resmgmt.WithRetry(retry.DefaultResMgmtOpts))
+	if err != nil {
+		return fmt.Errorf("failed to install chaincode: %v", err)
+	}
+	packageID := lifecyclepkg.ComputePackageID(installCCReq.Label, installCCReq.Package)
+	logger.Infof("%s successfully installed: %v", cc.Id, resp)
+
+	// approveformyorg
+	approveCCReq := resmgmt.LifecycleApproveCCRequest{
+		Name:              cc.Id,
+		Version:           cc.Version,
+		PackageID:         packageID,
+		Sequence:          cc.Seq,
+		EndorsementPlugin: cc.Escc,
+		ValidationPlugin:  cc.Vscc,
+		SignaturePolicy:   cc.Policy,
+		InitRequired:      cc.InitRequired,
+	}
+	txid, err := client.LifecycleApproveCC(nw.ChannelID, approveCCReq,
+		resmgmt.WithRetry(retry.DefaultResMgmtOpts),
+		resmgmt.WithTargetEndpoints(nw.Peers...),
+		resmgmt.WithOrdererEndpoint(nw.Orderers[0]),
+	)
+	if err != nil {
+		return fmt.Errorf("failed to approve chaincode: %v", err)
+	}
+	logger.Infof("%s successfully approved with txid: %v", cc.Id, txid)
+
+	// checkcommitreadiness
+	checkCCReq := resmgmt.LifecycleCheckCCCommitReadinessRequest{
+		Name:              cc.Id,
+		Version:           cc.Version,
+		Sequence:          cc.Seq,
+		EndorsementPlugin: cc.Escc,
+		ValidationPlugin:  cc.Vscc,
+		SignaturePolicy:   cc.Policy,
+		InitRequired:      cc.InitRequired,
+	}
+	chkresp, err := client.LifecycleCheckCCCommitReadiness(nw.ChannelID, checkCCReq,
+		resmgmt.WithRetry(retry.DefaultResMgmtOpts),
+		resmgmt.WithTargetEndpoints(nw.Peers...),
+	)
+	if err != nil {
+		return fmt.Errorf("failed to check chaincode commit readiness: %v", err)
+	}
+	logger.Infof("%s readiness check: %v", cc.Id, chkresp)
+
+	// commit
+	commitCCReq := resmgmt.LifecycleCommitCCRequest{
+		Name:              cc.Id,
+		Version:           cc.Version,
+		Sequence:          cc.Seq,
+		EndorsementPlugin: cc.Escc,
+		ValidationPlugin:  cc.Vscc,
+		SignaturePolicy:   cc.Policy,
+		InitRequired:      cc.InitRequired,
+	}
+	_, err = client.LifecycleCommitCC(nw.ChannelID, commitCCReq,
+		resmgmt.WithRetry(retry.DefaultResMgmtOpts),
+		resmgmt.WithTargetEndpoints(nw.Peers...),
+		resmgmt.WithOrdererEndpoint(nw.Orderers[0]),
+	)
+	if err != nil {
+		return fmt.Errorf("failed to commit chaincode: %v", err)
+	}
+	logger.Infof("%s successfully committed", cc.Id)
+
+	// init enclave
+	initReq := fpcmgmt.LifecycleInitEnclaveRequest{
+		ChaincodeID:         cc.Id,
+		EnclavePeerEndpoint: nw.Peers[0], // define the peer where we wanna init our enclave
+	}
+
+	initTxId, err := client.LifecycleInitEnclave(nw.ChannelID, initReq,
+		resmgmt.WithRetry(retry.DefaultResMgmtOpts),
+		resmgmt.WithTargetEndpoints(nw.Peers...), // peers that are responsible for enclave registration
+		resmgmt.WithOrdererEndpoint(nw.Orderers[0]),
+	)
+	if err != nil {
+		return fmt.Errorf("failed to init enclave: %v", err)
+	}
+	logger.Infof("%s successfully initialized enclave: %v", cc.Id, initTxId)
```

Comment:
Please see this as an example how `LifecycleInitEnclave` is supposed to use.

Reply:
looks nice and uniform to the existing lifecycle events